modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -389,45 +388,44 @@ private void onElectedAsLeader(long term) {
                     }
                 });
 
-        raftServiceAfterJoin().thenCompose(this::pushClusterConfigToCluster);
-    }
-
-    private CompletableFuture<Void> pushClusterConfigToCluster(CmgRaftService service) {
-        return service.readClusterState()
-                .thenCompose(state -> {
-                    if (state == null) {
-                        LOG.info("No CMG state found in the Raft service");
-                        return completedFuture(null);
-                    } else if (state.clusterConfigurationToApply() == null) {
-                        // Config was applied or wasn't provided
-                        LOG.info("No cluster configuration found in the Raft service");
-                        return completedFuture(null);
-                    } else {
-                        LOG.info("Cluster configuration is found in the Raft service, going to apply it");
-                        return distributedConfigurationUpdater.updateConfiguration(state.clusterConfigurationToApply())
-                                .thenCompose(unused -> removeClusterConfigFromClusterState(service));
-                    }
-                });
+        raftServiceAfterJoin().whenComplete((service, e) -> {
+            if (e != null) {
+                LOG.error("Error when joining to the raft service", e);
+                updateDistributedConfigurationActionFuture.completeExceptionally(e);
+            } else {
+                service.readClusterState()
+                        .thenAccept(state -> {
+                            String configuration = state.clusterConfigurationToApply();
+                            if (configuration != null) {
+                                updateDistributedConfigurationActionFuture.complete(
+                                        new UpdateDistributedConfigurationAction(
+                                                configuration,
+                                                () -> removeClusterConfigFromClusterState(service)
+                                        ));
+                            } else {
+                                updateDistributedConfigurationActionFuture.cancel(true);
+                            }
+                        });
+            }
+        });
     }
 
     private CompletableFuture<Void> removeClusterConfigFromClusterState(CmgRaftService service) {
         return service.readClusterState()
                 .thenCompose(state -> {
```

Comment:
I think we should check that configuration is actually present in the cluster state (i.e. it is not null) and only then update it